Framework/Core/src/DataSpecUtils.cxx

```diff
@@ -330,10 +330,10 @@ MatcherInfo extractMatcherInfo(DataDescriptorMatcher const& top)
       }
       // For now we do not support extracting a type from things
       // which have an OR, so we reset all the uniqueness attributes.
-      if (action.node->getOp() == ops::Or || action.node->getOp() == ops::Xor) {
-        state.hasError = true;
-        return VisitNone;
-      }
+      //if (action.node->getOp() == ops::Or || action.node->getOp() == ops::Xor) {
+      //  state.hasError = true;
+      //  return VisitNone;
+      //}
```

Comment:
@ktf, I guess this was added to allow quick implementation without considering too many corner cases. Maybe we can solve this now. At some point we want to support this anyhow.

Reply:
yes, exactly. The problem is extracting a ConcreteDataType from anything which is not a query. We basically need to revisit all the places where DataSpecUtils::asConcreteDataType is used and see if we cannot change it to something that matches the whole query. In particular I think the place where we serialise the workflow needs to have a recursive parser / serialiser to serialise and read back the query.